frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -403,7 +411,10 @@ export type ResourceLinkProps = {
   onClick?: () => void;
 };
 
-export type UseK8sModel = (groupVersionKind?: GroupVersionKind) => [K8sModel, boolean];
+export type UseK8sModel = (
+  // use groupVersionKind type instead of DeprecatedK8sReference, as migrating towards usage a group/version/kind object instead of the '~' concat stings
+  groupVersionKind?: K8sResourceKindReference | K8sGroupVersionKind,
+) => [K8sModel, boolean];
```

Comment:
@spadgett as suggested have updated the other types above i.e introduced new prop rather then or

but here as these are arguments' positions so had let it be one arg which accepts `K8sResourceKindReference | K8sGroupVersionKind` as once we migrate `UseK8sModel` to SDK then the signature won't break just type will strict. Let me know your thoughts.